pandas/core/base.py

```diff
@@ -1201,7 +1205,13 @@ def _map_values(self, mapper, na_action=None):
             else:
                 values = self.values
 
-            indexer = mapper.index.get_indexer(values)
+            try:
```

Comment:
I would rather you test mapper.index.is_unique here

Reply:
Outdated. @vnlitvin provided a better solution